cosmwasm/deploy-contract.js

```diff
@@ -0,0 +1,404 @@
+    }
+
+    return {
+        admin_address: adminAddress,
+        gateway_address: gatewayAddress,
+        multisig_address: multisigAddress,
+        service_registry_address: serviceRegistryAddress,
+        voting_verifier_address: verifierAddress,
+        destination_chain_id: destinationChainID,
+        signing_threshold: signingThreshold,
+        service_name: serviceName,
+        chain_name: chainId,
+        worker_set_diff_threshold: workerSetDiffThreshold,
+        encoder,
+        key_type: keyType,
+    };
+};
+
+const makeInstantiateMsg = (contractName, config, chain) => {
+    const {
+        axelar: { contracts },
+    } = config;
+
+    const { [contractName]: contractConfig } = contracts;
+
+    const { codeId } = contractConfig;
+
+    if (!isNumber(codeId)) {
+        throw new Error('Code Id is not defined');
+    }
+
+    switch (contractName) {
+        case 'ServiceRegistry': {
+            if (chain) {
+                throw new Error('ServiceRegistry does not support chainNames option');
+            }
+
+            return makeServiceRegistryInstantiateMsg(contractConfig);
+        }
+
+        case 'Multisig': {
+            if (chain) {
+                throw new Error('Multisig does not support chainNames option');
+            }
+
+            return makeMultisigInstantiateMsg(contractConfig, contracts);
+        }
+
+        case 'Rewards': {
+            if (chain) {
+                throw new Error('Rewards does not support chainNames option');
+            }
+
+            return makeRewardsInstantiateMsg(contractConfig);
+        }
+
+        case 'ConnectionRouter': {
+            if (chain) {
+                throw new Error('ConnectionRouter does not support chainNames option');
+            }
+
+            return makeConnectionRouterInstantiateMsg(contractConfig, contracts);
+        }
+
+        case 'NexusGateway': {
+            if (chain) {
+                throw new Error('NexusGateway does not support chainNames option');
+            }
+
+            return makeNexusGatewayInstantiateMsg(contractConfig, contracts);
+        }
+
+        case 'VotingVerifier': {
+            if (!chain) {
+                throw new Error('VotingVerifier requires chainNames option');
+            }
+
+            return makeVotingVerifierInstantiateMsg(contractConfig, contracts, chain);
+        }
+
+        case 'Gateway': {
+            if (!chain) {
+                throw new Error('Gateway requires chainNames option');
+            }
+
+            return makeGatewayInstantiateMsg(contracts, chain);
+        }
+
+        case 'MultisigProver': {
+            if (!chain) {
+                throw new Error('MultisigProver requires chainNames option');
+            }
+
+            return makeMultisigProverInstantiateMsg(contractConfig, contracts, chain);
+        }
+    }
+
+    throw new Error(`${contractName} is not supported.`);
+};
+
+const deploy = async (options, chain, config) => {
```

Comment:
This function doesn't seem to put the script on hold waiting for user confirmation, like it is done for the equivalent script dedicated to evm smart contracts